quil-rs/src/program/error/mod.rs

```diff
@@ -26,6 +26,8 @@ pub use leftover::LeftoverError;
 pub use result::{disallow_leftover, map_parsed, recover};
 pub use syntax::SyntaxError;
 
+// TODO: Should this be refactored as a enum for any error that can occur while working with a
+// `Program`?
```

Comment:
Yeah, that is what the existing doc comment suggested.

Unfortunately this error turned out to be more tightly wound in a lot of existing code than I expected. Refactoring it would be a heavier lift than I want to shoehorn into this already hefty PR. Unfortunately, The generic parameter makes it incompatible with the pyo3 macros as they exist today so it does get in the way a bit here.

Reply:
Created #150 to tracj